commonutils/src/main/java/io/intrepid/commonutils/StringUtils.java

```diff
@@ -0,0 +1,504 @@
+     * @param string2 The second string.
+     * @return The distance between the two strings.
+     */
+    public static int getLevenshteinDistance(CharSequence string1, CharSequence string2) {
+        int len1 = string1.length() + 1;
+        int len2 = string2.length() + 1;
+
+        // the array of distances
+        int[] cost = new int[len1];
+        int[] newcost = new int[len1];
+
+        // initial cost of skipping prefix in String string1
+        for (int i = 0; i < len1; i++) cost[i] = i;
+
+        // dynamicaly computing the array of distances
+
+        // transformation cost for each letter in string2
+        for (int j = 1; j < len2; j++) {
+            // initial cost of skipping prefix in String string2
+            newcost[0] = j;
+
+            // transformation cost for each letter in string1
+            for (int i = 1; i < len1; i++) {
+                // matching current letters in both strings
+                int match = (string1.charAt(i - 1) == string2.charAt(j - 1)) ? 0 : 1;
+
+                // computing cost for each transformation
+                int cost_replace = cost[i - 1] + match;
+                int cost_insert = cost[i] + 1;
+                int cost_delete = newcost[i - 1] + 1;
+
+                // keep minimum cost
+                newcost[i] = Math.min(Math.min(cost_insert, cost_delete), cost_replace);
+            }
+
+            // swap cost/newcost arrays
+            int[] swap = cost;
+            cost = newcost;
+            newcost = swap;
+        }
+
+        // the distance is the cost for transforming all letters in both strings
+        return cost[len1 - 1];
+    }
+
+    /**
+     * Has the same function as {@link java.lang.String#indexOf(int, int)} but it works on CharSequences as well.
+     *
+     * @param string The string to check.
+     * @param c      The character to look for.
+     * @param start  The index to start at (must be 0 <= stat < string.length())
+     * @return The first index on or after 'start' on which the character has been found or -1 if not found.
+     */
+    public static int indexOf(@NonNull CharSequence string, char c, int start) {
+        for (int i = start; i < string.length(); i++) {
+            if (string.charAt(i) == c) {
+                return i;
+            }
+        }
+        return -1;
+    }
+
+    /**
+     * Has the same function as {@link java.lang.String#equals(Object)} but it works on CharSequences as well.
+     *
+     * @param string1 First string.
+     * @param string2 Second string.
+     * @return True if the two strings are equal.
+     */
+    public static boolean equals(CharSequence string1, CharSequence string2) {
+        return compare(string1, string2) == 0;
+    }
+
+    /**
+     * Has the same function as {@link java.lang.String#compareTo(String)} but it works on CharSequences as well.
+     *
+     * @param cs1 First string.
+     * @param cs2 Second string.
+     * @return -1, 0, or 1.
```

Comment:
As written, this returns negative, zero, or positive depending on the comparison, but not always -1, 0, or 1.